web/src/app/administration/components/workbasket-list/workbasket-list.component.spec.ts

```diff
@@ -0,0 +1,151 @@
+const orientationServiceMock = jest.fn().mockImplementation(
+  (): Partial<OrientationService> => ({
+    getOrientation: getOrientationFn,
+    calculateNumberItemsList: jest.fn().mockReturnValue(1920)
+  })
+);
+
+const getImportingFinishedFn = jest.fn().mockReturnValue(of(true));
+const importExportServiceMock = jest.fn().mockImplementation(
+  (): Partial<ImportExportService> => ({
+    getImportingFinished: getImportingFinishedFn
+  })
+);
+
+@Component({ selector: 'taskana-administration-workbasket-list-toolbar', template: '' })
+class WorkbasketListToolbarStub {
+  @Input() workbaskets: Array<WorkbasketSummary>;
+  @Input() workbasketDefaultSortBy: string;
+  @Output() performSorting = new EventEmitter<Sorting>();
+  @Output() performFilter = new EventEmitter<Filter>();
+}
+
+@Component({ selector: 'taskana-administration-icon-type', template: '' })
+class IconTypeStub {
+  @Input() type: ICONTYPES = ICONTYPES.ALL;
+  @Input() selected = false;
+}
+
+@Component({ selector: 'taskana-shared-spinner', template: '' })
+class SpinnerStub {
+  @Input() isRunning: boolean;
+}
+
+@Component({ selector: 'taskana-shared-pagination', template: '' })
+class PaginationStub {
+  @Input() page: Page;
+  @Input() type: String;
+  @Output() changePage = new EventEmitter<number>();
+  @Input() numberOfItems: number;
+}
+
+@Component({ selector: 'svg-icon', template: '' })
+class SvgIconStub {}
+
+describe('WorkbasketListComponent', () => {
+  let fixture: ComponentFixture<WorkbasketListComponent>;
+  let debugElement: DebugElement;
+  let component: WorkbasketListComponent;
+  let store: Store;
+  let actions$: Observable<any>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [NgxsModule.forRoot([WorkbasketState]), MatSnackBarModule, MatDialogModule],
+      declarations: [
+        WorkbasketListComponent,
+        WorkbasketListToolbarStub,
+        IconTypeStub,
+        SpinnerStub,
+        PaginationStub,
+        SvgIconStub
+      ],
+      providers: [
+        { provide: WorkbasketService, useClass: workbasketServiceMock },
+        { provide: OrientationService, useClass: orientationServiceMock },
+        { provide: ImportExportService, useClass: importExportServiceMock }
+      ]
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(WorkbasketListComponent);
+    debugElement = fixture.debugElement;
+    component = fixture.componentInstance;
+    store = TestBed.inject(Store);
+    actions$ = TestBed.inject(Actions);
+    fixture.detectChanges();
+  }));
+
+  it('should create component', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should dispatch SelectWorkbasket when selecting a workbasket', async((done) => {
+    component.selectWorkbasket('1');
+    let actionDispatched = false;
+    actions$.pipe(ofActionDispatched(SelectWorkbasket)).subscribe(() => {
+      actionDispatched = true;
+      expect(actionDispatched).toBe(true);
```

Comment:
see above

Reply:
fixed by calling the the subscription in a zip 
`     zip(actions$.pipe(ofActionDispatched(SelectWorkbasket))).subscribe(() => (actionDispatched = true));
`    `    component.selectWorkbasket('1');
`
   `     expect(actionDispatched).toBe(true);
`